coral/src/app/features/topics/details/settings/TopicSettings.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Alert,
+  BorderBox,
+  Box,
+  Button,
+  PageHeader,
+  Typography,
+  useToast,
+} from "@aivenio/aquarium";
+import { useState } from "react";
+import { useMutation } from "@tanstack/react-query";
+import { useNavigate } from "react-router-dom";
+import { deleteTopic } from "src/domain/topic";
+import { useTopicDetails } from "src/app/features/topics/details/TopicDetails";
+import { parseErrorMsg } from "src/services/mutation-utils";
+import { TopicDeleteConfirmationModal } from "src/app/features/topics/details/settings/components/TopicDeleteConfirmationModal";
+
+function TopicSettings() {
+  const { topicName, environmentId, topicOverview } = useTopicDetails();
+
+  const isTopicOwner = topicOverview.topicInfoList[0].topicOwner;
+  const showDeleteTopic = topicOverview.topicInfoList[0].showDeleteTopic;
+
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  const [showConfirmation, setShowConfirmation] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
+
+  const { mutate, isLoading } = useMutation(
+    (params: { deleteAssociatedSchema: boolean; remark: string | undefined }) =>
+      deleteTopic({
+        topicName,
+        env: environmentId,
+        deleteAssociatedSchema: params.deleteAssociatedSchema,
+        remark: params.remark,
+      }),
+    {
+      onSuccess: () => {
+        navigate("/topics");
+        toast({
+          message: "Topic deletion request successfully sent",
+          position: "bottom-left",
+          variant: "default",
+        });
+      },
+      onError: (error) => {
+        setErrorMessage(parseErrorMsg(error));
+      },
+      onSettled: () => {
+        setShowConfirmation(false);
+      },
+    }
+  );
+
+  function getDeleteDisabledInformation() {
+    const topicHasOpenACLRequest =
+      topicOverview.topicInfoList[0].hasOpenACLRequest;
```

Comment:
I would like to confirm that `topicInfoList` is always an array of 1 element. Or what the circumstances are when it can be more than 1. Because I think we rely on this `topicInfoList[0]` in a few difference spots 🤔 Maybe a comment would be welcome?

Maybe this should not be an array at all.

Reply:
Yeah, that's kind of messy. 

How about:

We create our own type `TopicDetails` (or `TopicOverview`) that includes all properties we need and add a formatter function in `domain` to transform the response we're getting in the one we want. 

That way, only functions etc. inside `domain` know what the response from the API looks like, everything else in Frontend works with exact the information we need for Frontend. If there are changes in the API later to make response more easy for us, we only have to change it in one place. 

This is actually the main idea behind `domain` - make "app" code more independent from the exact structure of the api.

Reply:
Yes, this sounds better ^^

Reply:
OK, will do! ⌨️

Reply:
You know what - since this change would touch some files again, how about we merge this as it is for now and I'll make a new PR with the changes?